File: Data Structures & Algorithms/reverse-bits/submission-0.py
class Solution:
    def reverseBits(self, n: int) -> int:
        

        res=["0"]*32

        i=1
        idx=0
        while i<=n:
            if i&n:
                res[idx]="1"
            i*=2
            idx+=1

        s="".join(res)
        return int(s,2)

File: Data Structures & Algorithms/generate-parentheses/submission-0.js
class Solution {
    /**
     * @param {number} n
     * @return {string[]}
     */
    generateParenthesis(n) {
        let stack=[]
        let res=[]
        function backtrack(open,close){
            if(open==close&&close==n){
                res.push(stack.join(""))
                return
            }
            if(open<n){
                stack.push("(")
                backtrack(open+1,close)
                stack.pop()
            }
            if(open>close){
                stack.push(")")
                backtrack(open,close+1)
                stack.pop()


            }
        }
        backtrack(0,0)
        return res


    }
}
